code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -0,0 +1,336 @@
+	if(wisp.loc == src)
+		user << "You release the wisp. It begins to bob around your head."
+		user.sight |= SEE_MOBS
+		icon_state = "lantern"
+		wisp.orbit(user, 20)
+
+	else
+		if(wisp.orbiting)
+			var/atom/A = wisp.orbiting
+			if(istype(A, /mob/living))
+				var/mob/living/M = A
+				M.sight &= ~SEE_MOBS
+				M << "The wisp has returned to it's latern. Your vision returns to normal." //This works
+
+			wisp.stop_orbit()
+			wisp.loc = src
+			user << "You return the wisp to the latern."
+			icon_state = "lantern-blue"
+
+/obj/item/device/wisp_lantern/New()
+	..()
+	var/obj/effect/wisp/W = new(src)
+	wisp = W
+	W.home = src
+
+/obj/item/device/wisp_lantern/Destroy()
+	if(wisp)
+		if(wisp.loc == src)
+			qdel(wisp)
+		else
+			wisp.home = null //stuck orbiting your head now
+	..()
+
+//Wisp Lantern
+/obj/effect/wisp
+	name = "friendly wisp"
+	desc = "Happy to light your way."
+	icon = 'icons/obj/lighting.dmi'
+	icon_state = "orb"
+	var/obj/item/device/wisp_lantern/home
+	luminosity = 7
+	layer = FLY_LAYER - 0.3
+
+/obj/item/device/warp_cube
+	name = "blue cube"
+	desc = "A mysterious blue cube."
+	icon = 'icons/obj/lavaland/artefacts.dmi'
+	icon_state = "blue_cube"
+	var/obj/item/device/warp_cube/linked
+
+
+//Red/Blue Cubes
+
+/obj/item/device/warp_cube/attack_self(mob/user)
+	if(!linked)
+		user << "[src] fizzles uselessly."
+	if(linked.z == CENTCOMM)
+		user << "[linked] is somewhere you can't go."
+
+	PoolOrNew(/obj/effect/particle_effect/smoke, user.loc)
+	user.forceMove(get_turf(linked))
+	PoolOrNew(/obj/effect/particle_effect/smoke, user.loc)
+
+/obj/item/device/warp_cube/red
+	name = "red cube"
+	desc = "A mysterious red cube."
+	icon_state = "red_cube"
+
+/obj/item/device/warp_cube/red/New()
+	..()
+	if(!linked)
+		var/obj/item/device/warp_cube/blue = new(src.loc)
+		linked = blue
+		blue.linked = src
+
+//Meat Hook
+/obj/item/weapon/gun/magic/hook
+	name = "meat hook"
+	desc = "Mid or feed."
+	fire_sound = "sound/magic/Staff_Change.ogg"
+	ammo_type = /obj/item/ammo_casing/magic/hook
+	icon_state = "hook"
+	item_state = "chain"
+	fire_sound = 'sound/weapons/batonextend.ogg'
+	max_charges = 1
+	flags = NOBLUDGEON
+	force = 18
+
+/obj/item/ammo_casing/magic/hook
+	name = "hook"
+	desc = "a hook."
+	projectile_type = /obj/item/projectile/hook
+	caliber = "hook"
+	icon_state = "hook"
+
+/obj/item/projectile/hook
+	icon_state = "hook"
+	icon = 'icons/obj/lavaland/artefacts.dmi'
+	pass_flags = PASSTABLE
+	damage = 25
+	armour_penetration = 100
+	damage_type = BRUTE
+	hitsound = 'sound/effects/splat.ogg'
+	weaken = 2
+	var/chain
+
+/obj/item/ammo_casing/magic/hook/ready_proj(atom/target, mob/living/user, quiet, zone_override = "")
+	..()
+	var/obj/item/projectile/hook/P = BB
+	spawn(1)
+		P.chain = P.Beam(user,icon_state="chain",icon = 'icons/obj/lavaland/artefacts.dmi',time=1000, maxdistance = 30)
+
+/obj/item/projectile/hook/on_hit(atom/target)
+	. = ..()
+	if(isliving(target))
+		var/mob/living/L = target
+		L.forceMove(get_turf(firer))
```

Comment:
yeah okay throwing it then teleporting to the X is okay I guess
you can call it a reference